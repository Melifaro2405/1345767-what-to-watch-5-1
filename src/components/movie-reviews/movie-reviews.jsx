import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {reviewProptypes} from "../../props-validation";
import ReviewsList from "../reviews-list/reviews-list";
import {connect} from "react-redux";
import {fetchReviews} from "../../serviсes/api-actions";

class MovieReviews extends PureComponent {

  componentDidMount() {
    const {loadReviews, id} = this.props;
    loadReviews(id);
  }

  componentDidUpdate(prevProps) {
    const {loadReviews, id} = this.props;
    if (id !== prevProps.id) {
      loadReviews(id);
    }
  }

  render() {
    const {reviews} = this.props;
    const evenReviews = reviews.filter((_, index) => index % 2 !== 0);
    const oddReviews = reviews.filter((_, index) => index % 2 === 0);

    return (
      <div className="movie-card__reviews movie-card__row">
        <div className="movie-card__reviews-col">
          <ReviewsList reviews={oddReviews}/>
        </div>
        <div className="movie-card__reviews-col">
          <ReviewsList reviews={evenReviews}/>
        </div>
      </div>
    );
  }
}

MovieReviews.propTypes = {
  id: PropTypes.number.isRequired,
  reviews: PropTypes.arrayOf(PropTypes.shape(reviewProptypes)).isRequired,
  loadReviews: PropTypes.func.isRequired
};

const mapStateToProps = ({DATA}) => ({
  reviews: DATA.comments
});

const mapDispatchToProps = (dispatch) => ({
  loadReviews(id) {
    dispatch(fetchReviews(id));
  }
});

export {MovieReviews};
export default connect(mapStateToProps, mapDispatchToProps)(MovieReviews);
