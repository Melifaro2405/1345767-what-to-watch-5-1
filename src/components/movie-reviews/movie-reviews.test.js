import React from 'react';
import renderer from 'react-test-renderer';
import {MovieReviews} from './movie-reviews';

const noop = () => {};

export const reviews = [
  {
    comment: `A movie that will take you to another world full of emotions.`,
    date: `2020-11-03T13:38:44.769Z`,
    id: 1,
    rating: 8.6,
    user: {
      id: 11,
      name: `Jack`
    }
  }
];

it(`Should MovieReviews render correctly`, () => {
  const tree = renderer
    .create(
        <MovieReviews
          id={1}
          reviews={reviews}
          loadReviews={noop}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
