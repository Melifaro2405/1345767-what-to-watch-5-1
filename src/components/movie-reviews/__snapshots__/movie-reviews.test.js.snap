// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieReviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          A movie that will take you to another world full of emotions.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Jack
          </cite>
          <time
            className="review__date"
            dateTime="2020-11-03"
          >
            November 03, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.6
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  />
</div>
`;
